server/src/shared/Repositories/tags.repository.ts

```diff
@@ -0,0 +1,89 @@
+// Other dependencies
+import { Repository, EntityRepository } from 'typeorm'
+
+// Local files
+import { TagsEntity } from '../Entities/tags.entity'
+
+@EntityRepository(TagsEntity)
+export class TagsRepository extends Repository<TagsEntity> {
+
+    async getTrendingTags(): Promise<{ tags: TagsEntity[], count: number }> {
+        const [tags, total] = await this.findAndCount({
+            where: {
+                updated_at: {
+                    // Last 24 hours
+                    $gte: new Date(new Date().setDate(new Date().getDate() - 1))
+                }
+            },
+            order: {
+                popularity_ratio: 'DESC',
+                updated_at: 'DESC'
+            },
+            take: 30,
+            skip: 0,
+        })
+
+        return { tags, count: total }
+    }
+
+    async searchTag(searchValue: string): Promise<{ tags: TagsEntity[] }> {
+        const [tags] = await this.findAndCount({
+            where: {
+                name: new RegExp(searchValue, 'i')
+            },
+            take: 25,
+            order: {
+                popularity_ratio: 'DESC'
+            }
+        })
+
+        return { tags }
+    }
+
+    async tagActionOnTitleCreateOrUpdate(tagName: string): Promise<void> {
+        const tag = await this.findOne({ name: tagName })
+
+        if (!tag) {
+            this.save({
+                name: tagName,
+                popularity_ratio: 0.01,
+                total_title: 1
+            })
+        }
+
+        else {
+            // @ts-ignore
+            const diff = new Date().getTime() - tag.updated_at
+            const diffAsMin = Math.round((diff / 1000) / 60)
+
+            if (diffAsMin <= 1440 && tag.popularity_ratio < 1.00) tag.popularity_ratio += 0.01
+            else tag.popularity_ratio = 0.02
```

Comment:
the 1440 means the minutes in a day = 24 hours * 60 mins. The same, we could use the magic number 86400000, DAY_IN_MS, here to make the code readable:
```suggestion
            const diff = new Date().getTime() - tag.updated_at
            if (diff <= DAY_IN_MS && tag.popularity_ratio < 1.00) tag.popularity_ratio += 0.01
            else tag.popularity_ratio = 0.02
```